lib/src/validator/size.dart

```diff
@@ -21,8 +24,21 @@ class SizeValidator extends Validator {
     return packageSize.then((size) {
       if (size <= _MAX_SIZE) return;
       var sizeInMb = (size / math.pow(2, 20)).toStringAsPrecision(4);
-      errors.add("Your package is $sizeInMb MB. Hosted packages must be "
-          "smaller than 100 MB.");
+      // Current implementation of Package.listFiles skips hidden files
+      var ignoreExists = fileExists(join(entrypoint.root.dir, '.gitignore'));
+
+      var error = new StringBuffer(
+          "Your package is $sizeInMb MB. Hosted packages must be smaller than 100 MB.");
+
+      if (ignoreExists && !entrypoint.root.isUnderSourceControl) {
+        error.write(
+            " Your .gitignore has no effect since your project does not appear to be in version control.");
+      } else if (!ignoreExists && entrypoint.root.isUnderSourceControl) {
+        error.write(
+            " All files were included in the package since your project does not have .gitignore.");
```

Comment:
English is not my native language so I suppose this can be written better.

Reply:
I'd write "Consider adding a .gitignore to avoid including temporary files."